### PR TITLE
Medical Dispatch POI

### PR DESCRIPTION
**Pull request by dustylens**
## About the PR

This PR adds a new POI to serve as a shipyard for medical craft and a drop off point for medical bounties. 

It also serves as an initial base for the Medical Operations Manager (MOM). A new whitelisted role that will serve as a testbed for rolling out medical pay systems in addition to serving as a public face for organizing medical play and incorporating new players into the field.

## Why / Balance

- Maintainer request for a new place to store medical ships and bounty drop off points.
- Maintainer request for a place to stash a new medical director to facilitate pay.

As per request this POI offers virtually no specific services and is not, in fact, a medical center. It is a place where players may spawn in, purchase a ship and vital round start equipment and then go on their merry way. It is a gathering place for medics and a good space to find and collaborate with peers. But it is not a hospital in its default form.

The whitelisted role is currently intended to serve as a guide and to facilitate cooperation by example and via provision of basic pay to selected cooperating medics. We're going to feel this one out as time goes on and adjust based on feedback and experience.

There will be one shift Mom, but the medic roles themselves remain pilot/mercenary picks.

Currently the station has spawn in points for pilots, mercenaries and the MOM (love you mom). 

There is no contractor spawn in point. This seemed appropriate as it might be more than a bit of a noob trap. Players who want to sign on with medical crews may do so through traditional means.

Mom access is currently based on the SR. This could probably be dropped to the STC based on ongoing discussion.
Mom loadouts are currently based on mercenary loadouts due to lack of specific role drip. That isn't something that I view as a blocker at present and not worth holding up the PR. CMO gear is provided in the spawn in room.

This PR is, I feel, currently at an acceptable MVP state and is worth examining for tweaks and changes to ensure that naming conventions, role permissions and other requirements are where they should be. This is a first run for me so I may have missed a few things.

The biggest to:do on my own end is largely aesthetics based. Which is going to go about as quickly as my own spiriting skills permit.

There are a lot of role and loadout changes here that are a first for me. So I anticipate that there will be some initial bugs to iron out. 

## How to test

Fire it up. Try spawning in. Check out the station. Check out the roles. 

## Media

ss+(2024-11-23+at+04 00 11)
ss+(2024-11-23+at+04 01 59)
ss+(2024-11-23+at+04 01 48)

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

🆑 
- add: Added Medical Dispatch, a POI with spawn points.
- add: Added a new whitelisted command role, the Director of Care.
- tweak: Medical ships moved from Frontier to new medical dispatch POI.

### CODE REVIEW

**Comment by whatston3** [comment]
For what it's worth, I've written up a medical bank account, started "taxing" medical bounties (50% for now), added a modest stipend (18k/hr).  Replaced the mapped station admin console with a medical version - should be functionally set up for the new role.  All changes up on [whatston3:2024-11-28-NF-Dispatch-2-bank-n-bounty](https://github.com/new-frontiers-14/frontier-station-14/compare/master...whatston3:frontier-station-14:2024-11-28-NF-Dispatch-2-bank-n-bounty?expand=1)

**Comment by dvir001** [comment]
image

**Comment by BlueHNT** [comment]
Station with Ops manager, might add some points system for recovered bodies from those emergencies rather than the money. So, the manager can buy advanced topicals so you can focus on the actual healing, rather than spending 2 hours in space making a few stitches and whatnot.

**Comment by whatston3** [comment]
Minor point - in this hallway here, the radar consoles currently get a rotated view of the station, showing south as "up".  Might recommend swapping the position of the poster/map (doesn't have the same issue) and the radar consoles.  Nice use of wallmount radar consoles, by the way.

image

Might give the external airlocks dock numbers, but that can happen later.

I think the command blue makes sense for the job icon - the existing CMO icon seems ideal to me: denotes a position of command and associates with medical personnel via the cross, and I'd argue the ID card should be the same silver that the STC has (might give the SR/Sheriff gold IDs FWIW), presumably with a medical-coloured light blue insert along the bottom.

**Comment by dvir001** [comment]
Really need someone to review the map, the code side is ready.

**Review by dvir001 — APPROVED** [review]
On my way to buy robux
image

Code side all good, require map pass.

**Comment by MagnusCrowe** [comment]
**There's too many lights. Pretty much across the entire station.  Here are a series of recommendations.**
Morgues: I used a single blue light to improve the "morgue ambience" and I replaced two of the morgue slabs with a scanner computer and an operating table...two props that are at home in the morgue, both offering visual contrast.
image
Docks: Letting more of that green and red light in from outside the hallways and lowering the overall brightness throughout.
image
Gardens:  I recommend using glowing plant life to replace some of the harsh bright tubes with some softer colored lights.  I put the tall tree on the side without the table to prevent rotations from hiding the table.
image
CMO Office, bridge, and south vending hall: These were mostly ok, I think I moved a light in the hallway around the corner and deleted the light over the computers in the bridge, allowing the light they give off to be seen better.
Construction area: Used to have 3 lights, replaced with a single sodium light.  Players can replace it or build new light fixtures themselves.  The sodium light gives off the "unfinished construction" vibe better than a big blast of white light.
image
Bar/Lounge: I replaced all the lights in this room with 2 orange lights.  It had a lot of cozy furniture and flooring, but very clinical and sterile lighting.  The orange lights fit the cozy furniture vibe and since there's only 2 of them, it's not overpowering (and it fits the fireplace light better).
image
Here's what some areas look like from a player perspective.
image
image
image
image
image
image

**Comment by MagnusCrowe** [comment]
image
What about these areas are "Cargo"? 
image

image
Good job having the shuttle arrive at the station.

image
This room confuses me.  It's got treatment beds and cryo beds and a table and charger.  Is it a treatment room?  It should be bigger, have a medical locker, and maybe more table space. Is it a cryo bed room?  Why does it have all this extra treatment stuff in it.  It seems cramped and unfocused.

image
Janitorial/disposals looks good, only question is, why does it have a gas condenser? 

image
I don't think a few gas cans are going to be enough to fill this station in the event of a loss of atmosphere.  I would recommend adding gas tank chambers like you see on the trade station.  They shouldn't have miners in them but they can have atmos marker tiles placed on them, these types of tanks hold an immense amount of gas. 
image
VS 
image

**Comment by MagnusCrowe** [comment]
> Thanks for the walk through.
> 
> I'm unsure about the trees in the garden space. But I'll play around with the area to see what else might be appropriate.
> 
> I originally had a much greater variety in lighting choices at spawn but ultimately abandoned that approach when most of the lights would be burnt out when the POI spawned in. But we have a new box of colored tubes thanks to another recent PR so that might be somewhat practical now.
> 
> This was the reason for creating the box of red/green bulbs for this PR, as otherwise I was finding that many of the dock lights were MIA.

Yeah the lights being busted is annoying, I think we should probably have a spawning flag to disable that on spawn.  The lights being busted by solar flares could also be toned down a little IMO.

**Comment by MagnusCrowe** [comment]
Remove the anti-anomaly zone.

**Comment by MagnusCrowe** [comment]
> > Remove the anti-anomaly zone.
> 
> Sure.
> 
> > What about these areas are "Cargo"?
> 
> Easily change to dock if that seems more appropriate to you.
That area is a docking hub right? It seems more appropriate to me for it to be labeled as that.  It doesn't seem to have any connection to cargo at all.  It doesn't hold cargo, it doesn't sell cargo, it doesn't buy cargo.
May also consider calling it a lobby?  You can add a new station beacon for that label if you like.

> > Why does it have a gas condenser?
> 
> Largely unmapped tool due to the power draw that that was added as an additional utility piece for anyone who might have need of it. Which, it turns out, are a lot of visiting medics.
Yeah I understand it's a useful tool but why is it in the janitor closet?  It's a chemist machine.  I don't see the connection for why a janitor or someone doing cleaning would need a gas condenser.

> > I don't think a few gas cans are going to be enough to fill this station in the event of a loss of atmosphere. I would recommend adding gas tank chambers like you see on the trade station.
> 
> I think this was advised against when the station was originally being put together but persons siphoning atmosphere has largely subsided after the atmos price adjustments. Seems reasonable.

Siphoning isn't a really big deal if you don't include the miner like I specified.

**Comment by MagnusCrowe** [comment]
> > May also consider calling it a lobby?
> 
> Could do. But as there's no lobby station beacon wouldn't that qualify as giving it a custom label? Just want to be sure that's good to go.

Yeah, the only reason you have a shops beacon is because I made it custom for the trade station.  Go nuts.  Now that I think of it, "Lobby" is a pretty useful beacon on Frontier.  Or "Hub".

> > Yeah I understand it's a useful tool but why is it in the janitor closet?
> 
> _gestures vaguely to the POI having a mandate to be small and spare_ General utility space. Reclaimer for materials, reclaimer for gas. Hook up a scrubber. If the MOM wants to move it elsewhere that's a choice. It's basically in the utility closet.

Considering it a "reclaimer" for gas is not a good argument because your station has no closed waste gas loop, all waste gases are vented, nothing can be reclaimed from it.  I'd be more inclined to believe your argument if it were directly connected to the waste pipe or at least had a canister port on the waste pipe to collect waste gases to be liquified after the canister fills.

> > Siphoning isn't a really big deal if you don't include the miner like I specified.
> 
> I believe I plumbed it up as desired.

You did.  Replace the walls facing into the engineering room with reinforced glass and grille. There should also be a selection of atmos signs that you could look at.  One of them should fit this area to indicate what the deal is with the little chambers.

**Comment by MagnusCrowe** [comment]
image
Include a box of normal light tubes and a box of normal light bulbs to mitigate the round start broken lights.

Your disposals pipe bounces garbage randomly when the conveyor isn't on.  I wonder if there isnt a better way to handle this.  Perhaps changing the angle of the disposals pipe or putting a windoor in.
image

**Comment by MagnusCrowe** [comment]
These are comments about the new job:

MOM, really? Going with MOM? I swear to god if anyone calls me Mommy when I'm playing this job, I'm blaming you, not them.

I suggest making a locker specific to this role, making it indestructible and un-stealable (like the SR locker) so that we can put some job specific tools into it like maybe a full hypospray.

I strongly disagree with this job having access to the SR locker and everything else SR.  If necessary, it should have it's own access IMO.
image


image
Suggestions for the guidebook:

**Shift Objectives**
1) Pay attention to medical implant alerts and coordinate the paramedic response.  Ensure paramedics are routed to alerts and follow-up with patients to make sure they were cared for.
2) Coordinate medical training/mentorship by pairing interns with doctors.  You may personally train interns as long as it does not interfere with your other duties.
3) Manage the medical dispatch center's budget.  Compensate medical teams for their work when patients refuse to pay for service.  You are entitled to 20,000 spesos per hour worked from the dispatch center's budget.

**Medical Dispatch Budget** (not "your" ledger)
You will find a small medical dispatch budget at your Station Administration Console [insert picture].  This fund is replenished by taking a cut of each successful medical bounty claimed in this sector.

You may spend this money:
- To pay your own 20,000 speso per hour salary.
- To pay for station expenses. (Janitor services, hospitality services, upgrades, equipment, etc.)
- To compensate medical crews that heal patients who refuse to pay or cannot pay for medical services.

I would reword the last sentence in the guidebook entry to make it clearer that you can leave the dispatch center only as long as your responsibilities are being fulfilled.

**Comment by MagnusCrowe** [comment]
> > MOM, really? Going with MOM? I swear to god if anyone calls me Mommy when I'm playing this job, I'm blaming you, not them.
> 
> Remind me to add a bar of soap to Mom's loadout inventory.

Director of Care (DoC) is my suggestion.

**Comment by Temoffy** [comment]
DoC's not bad, but a large majority of SS14 already has MOM as a head position.

**Comment by arimah** [comment]
> DoC's not bad, but a large majority of SS14 already has MOM as a head position.

Really? I thought they mostly used Chief Medical Officer.

**Comment by Temoffy** [comment]
Rouny's

: )

**Comment by arimah** [comment]
RMC calls it MOM as a homage to Alien's MU-TH-UR/"mother" computer, no? I'd honestly much rather go with something less likely to result in weird RP. DoC seems like a good alternative, even if I'm not looking forward to the "what's up, DoC?" jokes. Alternatively stick with CMO, if this person is meant to be the premier authority on healthcare in the sector.

**Comment by Temoffy** [comment]
I was attempting to joke about the xeno queen being a head role, but I failed.

I think any of the three names could work, but I see the potential for uncomfortable situations and comments for MoM.

**Comment by MagnusCrowe** [comment]
> Some unique apparel and the like would be ideal. Both to keep the current CMO stuff in broad circulation but to also differentiate the role. However making visually interesting clothing in a timely manner is not something that I've had much success with. If a real spriter wanted to take on the task I'd be happy to do the backend work.

I was also thinking of giving them a medical ship voucher that they could use or give up to another player.

File: Resources/Prototypes/_NF/Catalog/Fills/Paper/forms_medical.yml [hunk 1]
```diff
@@ -178,3 +180,72 @@
...
+  parent: Paper
+  components:
+  - type: Paper
+    content: |2
+      [color=#52B4E9]◥[bold]N[/bold]◣ [bold]Medical Operations Manager Payroll[/bold][/color]
+      [color=#52B4E9]‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾[/color]
+      [head=2]Medical Service Invoice[/head]
+
+      [head=3]Receipt of service for compensation. [/head]
+
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Paper/forms_medical.yml:197]:
```suggestion
```

I don't think this really adds anything.

File: Resources/Prototypes/_NF/Entities/Objects/Devices/pda.yml [hunk 2]
```diff
@@ -441,8 +441,52 @@
   components:
   - type: Pda
     id: NFJanitorIDCard
+    state: pda-janitor
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Devices/pda.yml:444]:
```suggestion
```

File: Resources/Prototypes/_NF/Entities/Objects/Devices/pda.yml [hunk 3]
```diff
@@ -441,8 +441,52 @@
...
+    state: pda-mom
+  - type: CartridgeLoader
+    preinstalled:
+    - CrewManifestCartridge
+    - NotekeeperCartridge
+    - NewsReaderCartridge
+    - BountyContractsCartridge
+    - AstroNavCartridge    
+    - LedgerCartridge
+    - MedTekCartridge #medtek preloaded. Should be ok.
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Devices/pda.yml]:
```suggestion
    - MedTekCartridge #medtek preloaded. Should be ok.
  - type: Appearance
    appearanceDataInit:
      enum.PdaVisuals.PdaType:
        !type:String
        pda-mom
```

File: Resources/Prototypes/_NF/Entities/Structures/atm.yml [hunk 4]
```diff
@@ -227,6 +228,16 @@
...
   description: Used to transfer funds to and from NFSD's bank account.
+  suffix: NFSD
   components:
   - type: StationBankATM
     account: Nfsd
+
+- type: entity
+  parent: BaseStationAdminBankATM
+  id: StationAdminBankATMMedical
+  description: Used to transfer funds to and from NFSD's bank account.
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Structures/atm.yml:239]:
```suggestion
  description: Used to transfer funds to and from the medical bank account.
```

**Review by whatston3 — COMMENTED** [review]
Partial review.

File: Resources/Prototypes/_NF/Roles/Jobs/Medical/doc.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+- type: job
+  id: DirectorofCare
+  name: job-name-doc
+  description: job-description-doc
+  playTimeTracker: JobDirectorofCare
+  requirements:
+    - !type:OverallPlaytimeRequirement
+      time: 0 # 72000 # 20 hrs
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Roles/Jobs/Medical/doc.yml:8]:
```suggestion
      time: 72000 # 20 hrs
```
Shouldn't hurt.

File: Resources/Prototypes/_NF/PointsOfInterest/medical.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  spawnGroup: Required
+  gridPath: /Maps/_NF/POI/medical.yml
+  addComponents:
+  - type: IFF
+    color: "#00b7eb"
+    readOnly: true
+  - type: Shuttle
+    angularDamping: 999999
+    linearDamping: 999999
+  - type: StationTransit
```
Comment (whatston3) [inline Resources/Prototypes/_NF/PointsOfInterest/medical.yml:26]:
```suggestion
- type: pointOfInterest
  id: Medical
  parent: BasePOI
  name: 'Medical Dispatch'
  minimumDistance: 1650
  maximumDistance: 3400
  spawnGroup: Required
  gridPath: /Maps/_NF/POI/medical.yml
  addComponents:
  - type: IFF
    color: "#00b7eb"
  - type: StationTransit
```
A few of these fields can be inherited from the parent.
Not 100% sure these changes work, but should be roughly good.

File: Resources/Prototypes/_NF/Loadouts/role_loadouts.yml [hunk 7]
```diff
@@ -204,6 +204,27 @@
...
+  - ContractorGlasses
+  - DocPDA
+  - ContractorBelt
+  - DocEars
+  - ContractorBoxSurvival
+  - ContractorCartridge
+  - ContractorImplanter
+  - ContractorFun
+  - ContractorTrinkets
+  - ContractorBureaucracy
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Loadouts/role_loadouts.yml:226]:
```suggestion
  - ContractorBureaucracy
  - NFSpeciesSpecific
```
If you'd prefer the vox to have a medical mask by default, I can get you an NFMedicalSpeciesSpecific definition

File: Resources/Prototypes/_NF/Loadouts/doc_loadout_groups.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  name: loadout-group-contractor-shoes
+  loadouts:
+  - MercenaryClothingShoesBootsMercenary
+  - MercenaryClothingShoesBootsJack
+  - MercenaryClothingShoesBootsCombat
+  subgroups:
+  - PilotShoes
+  - ContractorShoes
+  fallbacks:
+  - MercenaryClothingShoesBootsMercenary
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Loadouts/doc_loadout_groups.yml:64]: Might suggest a loadout with black, brown, white shoes, maybe laceups?  Dunno.

File: Resources/Locale/en-US/_NF/job/job-description.ftl [hunk 9]
```diff
@@ -5,6 +5,7 @@ job-description-contractor = Fulfill any profitable contracts out on the frontie
 job-description-deputy = Catch criminals and enemies of the corporation, enforce the law, and ensure that the sector does not fall into disarray.
 job-description-nf-detective = Investigate crime scenes using forensic tools, ensure that the guilty party is found, and have a couple smokes.
 job-description-ertmailcarrier = Nothing stops the mail.
+job-description-medical-operations-manager = Provide guidance and direction for shift medics and work to safeguard the health of Nanotrasen personnel.
```
Comment (whatston3) [inline Resources/Locale/en-US/_NF/job/job-description.ftl:8]:
```suggestion
job-description-director-of-care = Provide guidance and direction for shift medics and work to safeguard the health of Nanotrasen personnel. 
```
MOM-to-DOC

File: Resources/Prototypes/_NF/Entities/Objects/Misc/rubber_stamp.yml [hunk 10]
```diff
@@ -81,3 +81,17 @@
   - type: Sprite
     sprite: _NF/Objects/Misc/stamps.rsi
     state: stamp-pal
+        
+- type: entity
+  name: Director of Care's rubber stamp
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Misc/rubber_stamp.yml:86]:
```suggestion
  name: director of care's rubber stamp
```
consistent casing

File: Resources/Prototypes/_NF/Entities/Objects/Misc/rubber_stamp.yml [hunk 11]
```diff
@@ -81,3 +81,17 @@
...
+  parent: RubberStampBase
+  id: RubberStampDoc
+  suffix: DO NOT MAP
+  components:
+  - type: Stamp
+    stampedName: stamp-component-stamped-name-doc
+    stampedColor: "#5b97bc"
+    stampState: "paper_stamp-nf-doctor"
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Misc/rubber_stamp.yml:94]:
```suggestion
    stampState: "paper_stamp-cmo"
```
I'd use this to be visually distinct from the doctor's stamp.

**Review by whatston3 — COMMENTED** [review]
minor stamp suggestions

**Review by whatston3 — APPROVED** [review]
I think this is good.

Changes I made that could use a double check:
* Optional POI count reduced to 6 from 8 (could probably even be reduced to 5)
* Medical forms reworked slightly - DOC forms have a separate header colour for easier visual ID (they're still close)
* DOC loadouts changed - they've got a new encryption key with command, medical, and common access and a separate centcom key (only because they report directly to them - if this changes to where the SR is their boss, which I could see an argument for, fair enough)
* "DirectorofCare" changed to "DirectorOfCare"
* Gas mixture on Medical Dispatch fixed

Things that require some acknowledgement:
* AllAccess is probably fine if it's a role that's at the same level as the SR - it's a whitelisted role, and gross abuse should be an administrative issue - the fact that the whitelisting process lets that through is its own problem.
* Ditto for centcomm radio - since centcomm is their boss, I think it makes sense
* The station admin console flatpack is to allow for the DOC, in cases where it is reasonable, to run with a ship of their choice without needing DOC-specific versions of the ship.  You want to run a Spirit for whatever reason?  Power to you, you can hand out payments from your ship without issue.

This looks good to me.

**Comment by Cheackraze** [comment]
https://github.com/new-frontiers-14/frontier-station-14/pull/2734 contains Medical Dispatch-specific holopads to be mapped at your descretion, 2 named ones, Reception and DoC's Office. Up to you how you use them, if they get merged before this one does in the next day or so

File: Resources/Prototypes/_NF/Catalog/Fills/Paper/forms_medical.yml [hunk 12]
```diff
@@ -133,12 +146,13 @@
 
           [bold]Additional Coverage:[/bold]
           \[  ] Towing: Coverage up to $10,000. Small shuttles only. ($2,000 premium)
+          \[  ] Towing: Coverage up to $20,000. Medium shuttles only. ($4,000 premium)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Paper/forms_medical.yml:149]:
```suggestion
          \[  ] Towing: Coverage up to $20,000. Small and medium shuttles only. ($4,000 premium)
```
Reasonable.

**Review by whatston3 — APPROVED** [review]

**Comment by dvir001** [comment]
@whatston3 https://www.youtube.com/watch?v=Rn6GbXZCt-o

**Comment by whatston3** [comment]
@dvir001
Gave my piece here: https://github.com/new-frontiers-14/frontier-station-14/pull/2451#pullrequestreview-2561065165
Seems good to go on my side.
Wrote up an intro blurb with Dusty that seems fine to me.

**Comment by Cheackraze** [comment]
i dont think we are at the limit of optional POIs anyhow since most of these being introduced lately are mandatory and shouldnt be bumping up against that limit. I also want to make a couple POIs to add a bit of variety in that specific spawn pool anyways.
